src/actions/mod.rs

```diff
@@ -812,6 +850,76 @@ impl ActionHandler {
     }
 }
 
+fn infer_config_defaults(project_dir: &Path, config: &mut Config) -> CargoResult<()> {
```

Comment:
Should this be in an `impl Config`? This would need more Cargo-related bits moved to `cargo.rs`, not sure if it's good from in terms of the module split/architecture